includes/Utilities/BackgroundJobCleanup.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace SkyVerge\WooCommerce\Facebook\Utilities;
+
+use SkyVerge\WooCommerce\Facebook\Heartbeat;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class BackgroundJobCleanup
+ *
+ * Responsible for cleaning up old completed background sync jobs from SkyVerge background job system.
+ * Each job is represented by a row in wp_options table, and these can accumulate over time.
+ *
+ * Note - this is closely coupled to the SkyVerge background job system, and is essentially a patch to improve it.
+ *
+ * @see SV_WP_Background_Job_Handler
+ *
+ * @since x.x.x
+ */
+class BackgroundJobCleanup {
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		$this->add_hooks();
```

Comment:
Personally I like to avoid coupling "real work" to class instantiation but not a blocker. A class as simple as this it's probably fine.

Reply:
If it's better to add this to an `init` hook, or something - let me know, happy to do that.

Reply:
.. I'm following the pattern used in `JobRegistry`. AKA `JobManager` (I think we should rename it, since it's not a generic registry).

Reply:
Perhaps, what are you basing your definition of a registry class on? I'm thinking it "registers" job classes and then "registers/adds" the job hooks for each class.

Reply:
I have a slight dislike for "manager" named classes because I've seen them end up as catch-all for all related functionality and then become a mess.

Reply:
A registry as I understand it is a generic container for well-known classes of things - so you might register logging classes, or middlewares for processing things, or image processing filters. Specifically (in our case), it doesn't know at compile time all the things that are in it. Any component that needs an X can get one from the registry. Any component that implements/provides an X registers it with the registry so others can get it.

https://www.martinfowler.com/eaaCatalog/registry.html

> I have a slight dislike for "manager" named classes because I've seen them end up as catch-all for all related functionality and then become a mess.

`Manager` and `Controller` both have that risk for sure. But I've now convinced myself that we shouldn't call this `JobRegistry` - let me know what you think. Unrelated to this PR so can follow up later :)

Reply:
Added a follow up issue for this! #1968